src/redux/session.js

```diff
@@ -103,3 +214,52 @@ module.exports.refreshSession = () => (dispatch => {
         return;
     });
 });
+
+module.exports.handleLogIn = (formData, callback) => (dispatch => {
+    dispatch(module.exports.setLoginError(null));
+    formData.useMessages = true; // NOTE: this may or may not be being used anywhere else
+    api({
+        method: 'post',
+        host: '',
+        uri: '/accounts/login/',
+        json: formData,
+        useCsrf: true
+    }, (err, body) => {
+        if (err) dispatch(module.exports.setLoginError(err.message));
+        if (body) {
+            body = body[0];
+            if (body.success) {
+                dispatch(module.exports.setLoginOpen(false));
+                body.messages.map(message => { // eslint-disable-line array-callback-return
+                    if (message.message === 'canceled-deletion') {
+                        dispatch(module.exports.setCanceledDeletionOpen(true));
+                    }
+                });
+                dispatch(module.exports.refreshSession());
+            } else {
+                if (body.redirect) {
+                    window.location = body.redirect;
+                }
+                // Update login error message to a friendlier one if it exists
+                dispatch(module.exports.setLoginError(body.msg));
+            }
+        }
+        // JS error already logged by api mixin
+        callback();
+    });
+});
+
+module.exports.handleLogOut = event => (dispatch => {
```

Comment:
While this should remain a thunk, I don't think we should have it handle the event. It should be used in `mapDispatchToProps` and that function should handle `event` and then dispatch this action.

Reply:
Hm. Several other of my hacky session.js functions do event.preventDefault() as well. Should I move all these out to components' callbacks?

I see the argument for event handling not belonging in a reducer, but I wonder if it's more brittle...